include/deal.II/base/mpi_noncontiguous_partitioner.templates.h

```diff
@@ -183,26 +183,45 @@ namespace Utilities
       const std::vector<types::global_dof_index> &indices_want,
       const MPI_Comm &                            communicator)
     {
+      // step 0) clean vectors from numbers::invalid_dof_index (indicating
+      //         padding)
+      std::vector<types::global_dof_index> indices_has_clean;
+      indices_has_clean.reserve(indices_has.size());
+
+      for (const auto &i : indices_has)
+        if (i != numbers::invalid_dof_index)
+          indices_has_clean.push_back(i);
+
+      std::vector<types::global_dof_index> indices_want_clean;
+      indices_want_clean.reserve(indices_has.size());
+
+      for (const auto &i : indices_want)
+        if (i != numbers::invalid_dof_index)
+          indices_want_clean.push_back(i);
+
       // step 0) determine "number of degrees of freedom" needed for IndexSet
       const types::global_dof_index n_dofs = Utilities::MPI::max(
         std::max(
-          [&indices_has]() {
-            const auto it = max_element(indices_has.begin(), indices_has.end());
-            return it == indices_has.end() ? 0 : (*it + 1);
+          [&indices_has_clean]() {
+            const auto it = std::max_element(indices_has_clean.begin(),
+                                             indices_has_clean.end());
+            return it == indices_has_clean.end() ? 0 : (*it + 1);
           }(),
-          [&indices_want]() {
-            const auto it =
-              max_element(indices_want.begin(), indices_want.end());
-            return it == indices_want.end() ? 0 : (*it + 1);
+          [&indices_want_clean]() {
+            const auto it = std::max_element(indices_want_clean.begin(),
+                                             indices_want_clean.end());
+            return it == indices_want_clean.end() ? 0 : (*it + 1);
```

Comment:
This would be easier to read if you avoided the lambdas inside `std::max` in favor of plain code here with temporary variable like `types::global_dof_index max_has = ...` and `max_wants`.

Reply:
I have made the change.